src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -646,6 +657,57 @@ internal void OnReaderCompleted(Action<Exception?, object?> callback, object? st
             }
         }
 
+        internal ValueTask<ReadResult> ReadAsync(int minimumBytes, CancellationToken token)
```

Comment:
The `ReadAsync()` path and `ReadAsync(minimumBytes)` path are not unified on purpose since there's quite a bit more overhead here:
- Checking the length 
- Potentially completing and scheduling the writer's awaitable